onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -199,6 +199,26 @@ struct OpenVINO_Provider : Provider {
 
     if (provider_options_map.find("load_config") != provider_options_map.end()) {
       load_config = provider_options_map.at("load_config");
+
+      // Enforce that the input path is absolute, reject if not
+      if (!std::filesystem::path(load_config).is_absolute()) {
+        throw std::invalid_argument("The config file path must be an absolute path: " + load_config);
+      }
+
+      auto resolve_path = [&](const std::string& path) -> std::string {
+        std::filesystem::path fs_path = path;
+        // Canonicalize the path to resolve symbolic links and remove '..' or '.'
+        try {
```

Comment:
It seems u r allowing the relative path here not rejecting it

Reply:
- The check for std::filesystem::path(filename).is_absolute() ensures that only absolute paths are accepted. Any relative paths (including those like ../..) are immediately rejected.

- The use of std::filesystem::canonical() ensures that the path is properly resolved (eliminating any . or .. references), ensuring that it points to the correct file. Ex. path = "../npu.json" resolves to "absolute/path/to/npu.json" here if its a valid location else you throw a runtime exception while resolving path.

Reply:
But it should not be allowed. What if it is pointing to a file with no access to current user ?

Reply:
@vthaniel can you please test this scenario,